lib/pry/method/weird_method_locator.rb

```diff
@@ -117,6 +123,13 @@ def find_method_in_superclass
         nil
       end
 
+      def prepended_method?(guess)
+        target_mod = @target.eval('self').class
+        target_mod.ancestors.take_while do |mod|
+          mod != target_mod
+        end.size > 0
```

Comment:
Nice code (re `take_while`)! but can replace this with `.any?` I think..

Reply:
It's not equivalent. Introduces two failures:
```
1) Pry::Method.from_binding should find the super method correctly
     Failure/Error: expect(m.owner).to eq a

       expected: #<Class:0x007ff166802c30>
            got: #<Class:0x007ff166802b40>

       (compared using ==)

       Diff:
       @@ -1,2 +1,2 @@
       -#<Class:0x007ff166802c30>
       +#<Class:0x007ff166802b40>
     # ./spec/method_spec.rb:130:in `block (3 levels) in <top (required)>'

  2) Pry::Method.from_binding should find the right method even if it was renamed and replaced
     Failure/Error: expect(m.source).to eq Pry::Method(o.method(:paella)).source

       expected: "def borscht\n  @nips = \"nips\"\n  binding\nend\n"
            got: "def borscht() paella end\n"
```

Reply:
@r-obert sorry i wasn't clear. I didn't mean replace the `take_while` i meant to replace the `size > 0` with `any?`

Reply:
oh. sure. i'll do that.

Reply:
done :)